src/Tools/dotnet-monitor/Egress/Extension/EgressExtension.cs

```diff
@@ -137,7 +168,7 @@ public void AddEnvironmentVariable(string key, string value)
                 StartInfo = pStart,
             };
 
-            using OutputParser<EgressArtifactResult> parser = new(p, _logger);
+            using OutputParser<EgressArtifactResult> parser = new(p, _logger, mode);
```

Comment:
Rather than updating `OutputParser` to understand modes and conditional logging, please try to pass `NullLogger<EgressExtension>.Instance` instead.